docs(DEVELOPER.md): explain how to get the Rust build system authenticated by GitHub (#13465)
Kong/kong: DEVELOPER.md

brew install libyaml
```

> Now, you have to authenticate with GitHub to download some essential repos.
> There are three ways to do it:

Reviewer: ```suggestion
Now, you have to authenticate with GitHub to download some essential repos
using either one of the following ways:
```

> * By [`gh`](https://cli.github.com/), basically `gh auth login`.

Reviewer: ```suggestion
* Download [`gh cli`](https://cli.github.com/) and run `gh auth login` once.
```

> * By [Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token). This token does not need to have any other permission than `Public Repositories (read-only)`, and set it as `GITHUB_TOKEN` environment variable.

Reviewer: ```suggestion
* Use a [Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token). This token does not need to have any other permission than `Public Repositories (read-only)`, and set it as `GITHUB_TOKEN` environment variable.
```

> * By [git credential helper](https://git-scm.com/docs/gitcredentials).

Reviewer: ```suggestion
* Use [git credential helper](https://git-scm.com/docs/gitcredentials).
```

> Then you have to make the rust build system also authenticate with GitHub,

Reviewer 2: ```suggestion
Then you have to make the Rust build system also authenticate with GitHub,
```

You will need to setup port forwarding on VirtualBox to be able to ssh into the box which can be done as follows:

1. Select the virtual machine you want to use and click "Settings"
2. Click the "Network" tab
3. Click the "Advanced" dropdown
4. Click "Port Forwarding"

> 5. Add a new rule in the popup. The only thing you will need is "Host Port" to be 22222 and "Guest Port" to be 22. Everything else can be left default (see screenshot below)

Reviewer 3: ```suggestion
5. Add a new rule in the popup. The only thing you will need is set "Host Port" to be 22222 and "Guest Port" to be 22. Everything else can be left default (see screenshot below)
```

Reviewer 2: You can use all `1.`s in Markdown and let the generator automatically maintain the sequence, it is a common practice and makes adding additional items easier.

Author: reverted

6. Click "Ok"

Now you should be able to `ssh <your_name>@127.1 -p 22222` to get SSH prompt. However, this requires us to type a long command and password every time we sign in. It is recommended you set up a public key and SSH alias to make this process simpler:

1. On your host machine, generate a keypair for SSH into the guest: `ssh-keygen -t ed25519`.
Just keep hitting Enter until the key is generated. You do not need a password for this key file since it is only used for SSH into your guest
2. Type `cat .ssh/id_ed25519.pub` and copy the public key
3. SSH into the guest using the command above
4. Create the ssh config directory (if it doesn't exist) `$ mkdir -p .ssh`
5. Edit the authorized keys list: `vim .ssh/authorized_keys`
6. Paste in the content of .ssh/id_ed25519.pub
7. Adjust the required privileges: `chmod 700 .ssh/`  and `chmod 400 .ssh/authorized_keys`
8. Logout of guest and make sure you are not promoted password when SSH again

> 9. Edit the .ssh/config file on your host and put in the following content:

Reviewer 2: Again, nothing to worry about leaving it all `1`s.

Author: reverted
